### PR TITLE
Adding generation of TypeScript definitions for the angular code.

### PR DESCRIPTION
**Pull request by alexkrauss**
This pull request continues the idea of the `ts` branch, by adding the ability to generate .d.ts files for the generated angular code. In particular, types are generated for the parameters and results of all methods, which allows interacting with an API in a statically typed fashion, including IDE support like code completion.

See the test for a simple example.

Also merged master into the ts branch.

### CODE REVIEW

**Comment by wcandillon** [comment]
Woah very cool. Is it possible to add a nodejs-typedef as well?

**Comment by alexkrauss** [comment]
@wcandillon I haven't looked at the node code yet, but I suppose it should be easy to add for this variant. Probably, the only difference will be the type of the returned promise...

**Comment by wcandillon** [comment]
@alexkrauss  yes, it's just about changing the return type.

**Comment by alexkrauss** [comment]
OK. Then we can simply make this another template variable. Will try to do this tonight or tomorrow.

**Comment by alexkrauss** [comment]
OK, as of 1ca4d3107e9f I added the node variant as well, as well as the possibility to add references to other typescript definitions (which is needed to resolve angular or node types).

Note that the result type `Q.Promise<{response: http.IncomingMessage, body: ...}>` for the node code just arised from me staring at the generated code and doing type inference in my head. I won't have time to actually write some typescript code against the generated stuff, so I cannot really test this right now. So I marked all this as experimental.

File: README.md [hunk 1]
```diff
@@ -246,6 +246,26 @@ eg: https://cloud.google.com/appengine/docs/go/requests#Go_Request_headers
...
+This includes the data types for the parameters and results of all methods. This allows static type checking of your
+API calls.
+
+TypeScript generation works similar to Javascript:
+```javascript
+var fs = require('fs');
+var CodeGen = require('swagger-js-codegen').CodeGen;
+
+var file = 'swagger/spec.json';
+var swagger = JSON.parse(fs.readFileSync(file, 'UTF-8'));
```
Comment (wcandillon) [inline README.md]: Can you add an example with `getTypeScriptDefinition` as well?

**Comment by alexkrauss** [comment]
The interface is really the same, so I just added a little note. Also removed the `dtsRefs`, which can actually be covered by the general `mustache` option.

**Comment by douglas-treadwell** [comment]
Is anything happening with this PR?

**Comment by alexkrauss** [comment]
@wcandillon Any chance of merging and releasing this change any time soon? I'd like to use this on a project where dependencies to unreleased versions from a git repository lead to technical and social challenges :-)

If I can help in any way, let me know.

**Comment by wcandillon** [comment]
@alexkrauss thks for this great contribution. As you will see, I have used your code partially for the typescript support.

**Pull request merged**